docs/Feature_guides/Real-time_updates.md

```diff
@@ -0,0 +1,188 @@
+
+- `gem 'graphql'`
+- working websockets (one is enough) -> GraphQL subscriptions use [ActionCable](https://guides.rubyonrails.org/action_cable_overview.html) internally so `cable.yml` needs to be configured
+
+## Implementation
+
+_PATTERN:_ Use multiple context-oriented subscriptions instead of a single big one. They're easier to manage and customize avoiding `if` conditions.\
+In our backlog board example we use 5 different subscriptions:
+
+- `easyProductBacklogBoard` (all actions not covered by the other ones)
+- `easyProductBacklogBoardInbox` (actions in inbox)
+- `pbiSplitter` (split action)
+- `easyProductBacklogBoardMovePbi` (reordering action)
+- `movePbiIntoSprintBoardSubscription` (moving into sprint board action)
+
+Further code examples will focus on `easyProductBacklogBoardMovePbi` subscription which handles actions that change positions of entities, product backlog items(PBIs) in this case.
+
+### Front-end
+
+General guide for implementing subscriptions via apollo-client - https://www.apollographql.com/docs/kotlin/tutorial/11-subscriptions
+and https://graphql-ruby.org/javascript_client/apollo_subscriptions.html.
+
+Subscriptions are defined in a single `subscriptions.ts` file and the graphql query is customized for each subscription.
+
+```ts
+// app/frontend/src/easy_agile/src/api/subscriptions.ts
+export const useProductBacklogBoardMovePbi = (easyProductBacklogBoardId: string) => {
+  return useSubscription(
+    gql`
+      subscription easyProductBacklogBoardMovePbi($easyProductBacklogBoardId: ID!) {
+        easyProductBacklogBoardMovePbi(easyProductBacklogBoardId: $easyProductBacklogBoardId) {
+          result {
+            mutationName
+            pbi {
+              id
+              name
+              color
+              icon
+              status
+              storyPoints
+              parent {
+                id
+                name
+                status
+              }
+              children: activeChildren {
+                id
+                name
+                color
+                icon
+                status
+                storyPoints
+                parent {
+                  id
+                }
+                prevId
+              }
+              issues {
+                id
+                subject
+              }
+              prevId
+            }
+            pbiFollower {
+              id
+              status
+              prevId
+            }
+            oldPbiFollower {
+              id
+              status
+              prevId
+            }
+          }
+        }
+      }
+    `,
+    {
+      easyProductBacklogBoardId,
+    }
+  );
+};
+```
+
+### Back-end
+
+##### Define the subscription class
+
+```ruby
+# plugins/easyproject/easy_plugins/easy_agile/api/easy_graphql/subscriptions/easy_product_backlog_board_move_pbi.rb
+module EasyGraphql
+  module Subscriptions
+    class EasyProductBacklogBoardMovePbi < ::EasyGraphql::Subscriptions::EasyProductBacklogBoard
```

Comment:
missing code from ::EasyGraphql::Subscriptions::EasyProductBacklogBoard .  

create an example that does not exist here, which will contain the code from the original subscription

```suggestion
    class EasyProductBacklogBoardMovePbi < ::EasyGraphql::Subscriptions::Base
    
   description 'Subscription for receiving updates about PBI position/status in EasyProductBacklogBoard'

      subscription_scope :easy_sprint_board_id
      argument :easy_sprint_board_id, ID, required: true
      
       field :result, Subscriptions::Results::MovePbiResult, null: true

      def subscribe(easy_sprint_board_id:)
        {
          easy_sprint_board: find_board(easy_sprint_board_id)
        }
      end

      private

      def find_board(id)
        ::EasySprintBoard.find_by(id: id)
      end
```

Reply:
Rewritten to include more details.